addons/google_gmail/models/google_gmail_mixin.py

```diff
@@ -11,6 +11,18 @@
 from odoo import _, api, fields, models, tools
 from odoo.exceptions import AccessError, UserError
 
+GMAIL_TOKEN_REQUEST_TIMEOUT = 5
+"""Timeout used when generating access token.
+The generated token is used until it's expiration (valid arround 1 hour)
+to open successive SMTP session
+"""
+GMAIL_TOKEN_VALIDITY_THRESHOLD = GMAIL_TOKEN_REQUEST_TIMEOUT + 1
+"""A threshold in seconds that we want to force the token
+to be re-generated because we are closed to its expiration time
+it gives a couple seconds of leeway to be safe to ensure we have
+time to open the new smtp session with the existing token.
+"""
+
```

Comment:
No need to explain everything here, just the reason it's defined that way (unless it's arbitrary, like the first one)
```suggestion
GMAIL_TOKEN_REQUEST_TIMEOUT = 5

# seconds removed from end-of-validity datetime to take into account the time
# needed to renew the token
GMAIL_TOKEN_VALIDITY_THRESHOLD = GMAIL_TOKEN_REQUEST_TIMEOUT + 5

```